libs/js/banglejs/E_showMenu_Q3.js

```diff
@@ -133,19 +133,19 @@
         l = g.setFont("6x15").wrapString(title,r.w-pad);
       g.setFontAlign(-1,0).drawString(l.join("\n"), r.x+12, r.y+H/2);
     },
-    select : function(idx) {
+    select : function(idx, touch) {
       if (idx<0) return back&&back(); // title
       var item = menu[keys[idx]];
       Bangle.buzz(20);
-      if ("function" == typeof item) item(l);
+      if ("function" == typeof item) item(l, touch);
       else if ("object" == typeof item) {
         // if a bool, just toggle it
         if ("number" == typeof item.value) {
           showSubMenu(item, keys[idx]);
         } else {
           if ("boolean"==typeof item.value)
             item.value=!item.value;
-          if (item.onchange) item.onchange(item.value);
+          if (item.onchange) item.onchange(item.value, touch);
```

Comment:
@gfwilliams should we also pass the scroller (`l`) here to mimic the capabilities of when the item is a function on L140?

It feels a little hacky/involved adding these things to `onchange`. Maybe there should be a separate function inside the `item` object named "func", "run" or similar that took the same role as if `item` had been a function?

Reply:
I'll do a small refactor and then ask you to comment when I pushed it to here.

Reply:
@gfwilliams 

For now I landed on not introducing a new extra function beside `onchange` inside entry objects. I believe that is better for backwards compatibility. 

(I go back and forth in my head on whether the extra function would be a good thing - please tell if you do like the idea)

Reply:
Yes, I think it's best not to introduce more functions unless you absolutely need it too.